src/main.jl

```diff
@@ -237,7 +235,11 @@ function SolverCore.solve!(
     set_dual_residual!(stats, nlp_at_x.current_score)
     set_iter!(stats, nlp_stop.meta.nb_of_stop)
     set_time!(stats, nlp_at_x.current_time - nlp_stop.meta.start_time)
-    callback(nlp, solver, stats)
+    if !success & (unsuccinarow >= max_unsuccinarow)
+      # don't just cycle if we have to many unsucessful iterations 
```

Comment:
I don' think the scenario, `success = true` and `unsuccinarow >= max_unsuccinarow` is possible.
```suggestion
    if unsuccinarow >= max_unsuccinarow
```